src/ComponentInstaller.php

```diff
@@ -187,12 +191,89 @@ public function onPostPackageInstall(PackageEvent $event)
                 return $injectors;
             }, new Collection([]))
             // Inject modules into configuration
-            ->each(function ($injector, $module) use ($name, $packageTypes) {
+            ->each(function ($injector, $module) use ($name, $packageTypes, $applicationModules) {
+                $injector->setApplicationModules($applicationModules);
```

Comment:
Your approach (a) is tested, and (b) solves the problem. There may be better solutions, but I'm satisfied with this one.